src/util.c

```diff
@@ -119,6 +122,157 @@ int find_yaml_glob(const char* rootdir, glob_t* out_glob)
     return 0;
 }
 
+static gboolean
+copy_yaml_subtree(yaml_parser_t *parser, yaml_emitter_t *emitter, GError** error) {
+	yaml_event_t event;
+    int map_count = 0, seq_count = 0;
+    do {
+		if (!yaml_parser_parse(parser, &event)) {
+            g_set_error(error, G_MARKUP_ERROR, G_MARKUP_ERROR_INVALID_CONTENT, "Error parsing YAML: %s", parser->problem);
+            return FALSE;
+        }
+
+        switch (event.type) {
+            case YAML_MAPPING_START_EVENT:
+                map_count++;
+                break;
+            case YAML_SEQUENCE_START_EVENT:
+                seq_count++;
+                break;
+            case YAML_MAPPING_END_EVENT:
+                map_count--;
+                break;
+            case YAML_SEQUENCE_END_EVENT:
+                seq_count--;
+                break;
+            default:
+                break;
+        }
+        if (emitter && !yaml_emitter_emit(emitter, &event)) {
+            // LCOV_EXCL_START
+            g_set_error(error, G_MARKUP_ERROR, G_MARKUP_ERROR_INVALID_CONTENT, "Error emitting YAML: %s", emitter->problem);
+            return FALSE;
+            // LCOV_EXCL_STOP
+        }
+    } while (map_count || seq_count);
+    return TRUE;
+}
+
+static gboolean
+drill_into_yaml_subtree(yaml_parser_t *parser, yaml_emitter_t *emitter, char** key, GError** error) {
```

Comment:
This function took me a while to grasp. It all makes sense, but I'd like to see a comment above briefly describing what this is supposed to do. Also stating that `char**keys` is not some kind of string to be returned to the caller (as is usual with this prototype), but rather a vector of the split up prefix.

Reply:
I added some comments, renamed the function to emit_yaml_subtree and the argument to yaml_path.

Reply:
Thank you, I like that much better!